intel_extension_for_transformers/llm/runtime/graph/core/layers/mha_dense.cpp

```diff
@@ -229,71 +239,60 @@ using ScaleWriteBackS32S8 = ScaleWriteBack<ISA_T, int32_t, int8_t>;
 
 /**
  * @brief PackedWeight(Default) with batch
- *
- * @tparam T element type of the weight
  */
-template <typename T>
-class PackedWeightBatch : public jblas::prologue::gemm::WeightBase, public jblas::prologue::ISerialBuffer {
+class StoragePackedWeightBatch : public jblas::storage::gemm::IWeightBase {
+  using Base = jblas::storage::gemm::IWeightBase;
+
  public:
-  explicit PackedWeightBatch(jblas::gemm::GemmCoreType _type) : jblas::prologue::gemm::WeightBase(_type), mBatch(0) {}
+  int mBatch;
+  jblas::storage::ObjectAlignedBuffer<NE_ALIGNMENT> mWBuf;
+  // size_t mWSize;
 
-  size_t resize(int NPad, int KPad, int num_batch) {
-    mNPad = NPad;
-    mKPad = KPad;
+  explicit StoragePackedWeightBatch(uint64_t _core_id) : Base(_core_id), mBatch(0) {}
+  size_t resize(int NPad, int KPad, int N, int K, int num_batch, JBLAS_DTYPE dtype) {
+    IWeightBase::resize(NPad, KPad, N, K, dtype);
     mBatch = num_batch;
-    auto size = (size_t)mBatch * NPad * KPad * jblas::gemm::getWeightSize(mCoreType);
-    jblas::prologue::ISerialBuffer::resize(size);
-    mSize = getSerializedSize();
+    auto bsize = static_cast<size_t>(mBatch) * NPad * KPad * jblas::utils::jblas_dtype_size(dtype);
+    mWBuf.resize(bsize);
+    mSize = utils::padto(IWeightBase::getSerializedSize() + mWBuf.getSerializedSize(), NE_ALIGNMENT);
     return mSize;
   }
 
-  virtual void assign(int8_t* buf) override {
-    jblas::prologue::gemm::WeightBase::deserializeBuffer(buf, true);
+  template <typename T>
+  inline constexpr T* WPtr() const {
+    return mWBuf.get<T>();
+  }
+
+  void assign(int8_t* buf) override {
     deserializeBuffer(buf, true);
-    jblas::prologue::ISerialBuffer::deserializeBuffer(buf, true);
-    mWPtr = get<T>();
-    mWSize = size<T>();
+    mWBuf.deserializeBuffer(buf, true);
   }
 
-  virtual void serialize(int8_t* wptr) {
-    jblas::prologue::gemm::WeightBase::serializeToBuffer(wptr);
+  void serialize(int8_t* wptr) override {
```

Comment:
need activation a serialization method? when will we serialize these?

Reply:
assert false looks better for me

Reply:
Will assert false in a following PR so that the implementation for the new jblas will in git history and (be restored easily if needed).